pages/index.js

```diff
@@ -1,30 +1,73 @@
-import React from "react";
+import React, { useRef, useState} from 'react';
 
 import TodoInput from "../components/todo-input";
 import TodoList from "../components/todo-list";
 
 import styled from "styled-components";
 
 export default function Home() {
+
+  const [todos, setTodos] = useState([]);
+  const [index, setIndex] = useState(1);
+
+  const addTodo = ({content, date}) => {
+    if ((content === undefined) || (date === undefined)) {
+      alert('모든 항목을 입력해주세요!');
+      return;
+    }
+
+    else if (String(date).length != 8){
+      alert('날짜를 올바른 형식으로 입력해주세요!');
+      return;
+    }
+
+    const todo = {
+      id: index,
+      content,
+      date,
+      checked: false
+    }
+    setTodos(todos.concat(todo)); 
+    setIndex(index + 1);
+    alert('입력 완료!')
+  }
```

Comment:
```suggestion
  const [todos, setTodos] = useState([]);

  const handleAdd = (todo) => {
    const { content, date } = todo;
    
    if (//input 값 하나라도 비어있으면) {
      alert('모든 항목을 입력해주세요!');
      return;
    }
    else if (String(date).length !== 8){
      alert('날짜를 올바른 형식으로 입력해주세요!');
      return;
    }

    const newTodo = {
      content,
      date
    }
    setTodos(todos.concat(newTodo)); 
    alert('입력 완료!');
  }
```
1. index를 state로 선언하지 않아도 delete가 잘 작동할 수 있게 만들 수 있습니다. ```components/todo-list.js```의 리뷰를 참고해주세요.
2. 함수 네이밍을 ```addTodo```로 하는 것보다 ```handleAdd```로 해주시는 게 좋아요. [참고](https://blog.sonim1.com/220)
3. 함수의 매개변수에 굳이 ```({content, date})``` 이렇게 써주실 이유가 없어요! 위에처럼 바꿔주시면 됩니다.
4. 처음 if문에서 ```content```와 ```date```가 ```undefined```인지 검사하고 있는데요,
페이지에 처음 접속해서 제출 버튼을 누르면 저 조건에 걸리지만
input과 textarea에 값을 입력했다가 지우면 undefined가 아니라 저 조건에 걸리지 않게 됩니다.
미션에 나온 조건대로 잘 실행되도록 로직 바꿔주셔야 할 것 같아요~
5. ```concat```할 ```todo```는 새로운 todo이기 때문에 ```newTodo```로 이름 바꿔주시면 좋을 것 같아요.
6. checked는 왜 넣으신 건가요?